ArchUnitNet.MSTest.CheckErrormessages/PlantUMLErrormessagesCheck.cs

```diff
@@ -0,0 +1,151 @@
+
+        private static string Filename { get; set; }
+
+
+        [TestMethod]
+        public void StartTest()
+        {
+            CheckByPuml(out string rawErrormessage);
+            //CheckForDuplications returns false when errormessage contains no duplications
+            bool isDuplications = !CheckForDuplications(rawErrormessage, out string explainErrormessage);
+
+            string errormessage = "\nOriginal (ArchUnitNet) Exception:\n" + rawErrormessage +
+                "\n\nExplained Error:\n" + explainErrormessage + "\n";
+
+            Assert.IsTrue(isDuplications, errormessage);
+        }
+
+        public bool CheckByPuml(out string errormessage)
+        {
+            errormessage = null;
+
+            try
+            {
+                IArchRule adhereToPlantUmlDiagram = Types().Should().AdhereToPlantUmlDiagram(Filename);
+                adhereToPlantUmlDiagram.Check(Architecture);
+            }
+            //xUnit
+            catch (FailedArchRuleException xUFAREx)
+            {
+                errormessage = xUFAREx.Message;
+
+                return false;
+            }
+
+            return true;
+        }
+
+        public bool CheckForDuplications(string uncutMessage, out string errormessage)
+        {
+            bool isDuplications = false;
+            errormessage = null;
+            StringBuilder message = new();
+
+            string[] errors = uncutMessage.Split('\n');
+            
+            if (CheckGroupsForDuplications(String.Empty, errors, out errors, out errormessage))
+            {
+                isDuplications = true;
+                message.Append(errormessage);
+            }
+
+            foreach (string error in errors)
+            {
+                if (!String.IsNullOrWhiteSpace(error))
+                {
+                    string[] ands = error.Split("and");
+                    string[] veryFirstError = ands[0].Split("does depend on");
+                    string firstError = veryFirstError.Length == 2 ? veryFirstError[1] : String.Empty;
+
+                    if (CheckGroupsForDuplications(firstError, ands, out ands, out errormessage))
+                    {
+                        isDuplications = true;
+                        message.Append(errormessage);
+                    }
+                }
+            }
+
+            errormessage = message.ToString();
+            if (String.IsNullOrWhiteSpace(errormessage))
+            {
+                errormessage = "\nNo Errors\n";
+            }
+
+
+            return isDuplications;
+        }
+
+        public bool CheckGroupsForDuplications(string firstError, string[] splittedMessages,
+            out string[] filteredMessages, out string errormessage)
+        {
+            errormessage = null;
+
+            var groups = splittedMessages.GroupBy(x => x.Trim().Trim('\t'));
+            var duplications = groups.Where(x => (x.Count() > 1 
+            && !String.IsNullOrWhiteSpace(x.FirstOrDefault()))
```

Comment:
Switch && and || to ensure Empty strings will be ignored.

!String.IsNullOrWhitespace(x.FirstOrDefaul) && (x.Count() > 1 || firstError.Contains(x.First())) 

This would protect from errors due Empty or Whitespace strings (depending on the string inside FirstError).

Reply:
Corrected